verticles/graphql/src/main/java/com/gentics/mesh/graphql/dataloader/DataLoaderKey.java

```diff
@@ -0,0 +1,58 @@
+package com.gentics.mesh.graphql.dataloader;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dataloader.DataLoader;
+
+import graphql.execution.ResultPath;
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * The GraphQL path-aware key for {@link DataLoader}.
+ * 
+ * @param <T>
+ */
+public final class DataLoaderKey<T> {
+
+	private final String path;
+	private final T value;
+
+	public DataLoaderKey(String path, T value) {
+		this.path = path;
+		this.value = value;
+	}
+
+	public DataLoaderKey(ResultPath path, T value) {
+		this(path.toString().replaceAll("\\[[\\d+]\\]", StringUtils.EMPTY), value);
```

Comment:
Result indices break the query paths, so cutting them off.

Reply:
Please add a comment about this in the Javadoc also.